strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java

```diff
@@ -235,7 +236,83 @@ private void applySchemaChanges(JanusGraphManagement jgm)
+                          jgm.getVertexLabel(ARTIFACT_TAG));
+
+        jgm.addConnection(jgm.getEdgeLabel(REMOTE_ARTIFACT_INHERIT_ARTIFACT),
+                          jgm.getVertexLabel(REMOTE_ARTIFACT),
+                          jgm.getVertexLabel(ARTIFACT));
+
+        jgm.addConnection(jgm.getEdgeLabel(ARTIFACT_COORDINATES_INHERIT_GENERIC_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(GENERIC_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(GENERIC_ARTIFACT_COORDINATES));
+
+        jgm.addConnection(jgm.getEdgeLabel(ARTIFACT_GROUP_HAS_ARTIFACTS),
+                          jgm.getVertexLabel(ARTIFACT_ID_GROUP),
+                          jgm.getVertexLabel(ARTIFACT));
+    }
+
+    private void applyPropertyConstraints(JanusGraphManagement jgm)
+    {
+        // Vertex Property Constraints
+        addVertexPropertyConstraints(jgm, ARTIFACT, "uuid", "storageId", "repositoryId", "created", "lastUpdated",
+                                     "lastUsed", "sizeInBytes", "downloadCount", "filenames", "checksums");
+
+        addVertexPropertyConstraints(jgm, REMOTE_ARTIFACT, "uuid", "cached");
+
+        addVertexPropertyConstraints(jgm, GENERIC_ARTIFACT_COORDINATES, "uuid", "version", "coordinates.extension",
+                                     "coordinates.name");
+
+        addVertexPropertyConstraints(jgm, RAW_ARTIFACT_COORDINATES, "uuid", "version", "coordinates.extension",
+                                     "coordinates.name", "coordinates.path");
+
+        addVertexPropertyConstraints(jgm, MAVEN_ARTIFACT_COORDINATES, "uuid", "version", "coordinates.extension",
+                                     "coordinates.name", "coordinates.groupId", "coordinates.artifactId",
+                                     "coordinates.classifier");
+
+        addVertexPropertyConstraints(jgm, NPM_ARTIFACT_COORDINATES, "uuid", "version", "coordinates.extension",
+                                     "coordinates.name", "coordinates.scope");
+
+        addVertexPropertyConstraints(jgm, NUGET_ARTIFACT_COORDINATES, "uuid", "version", "coordinates.extension",
+                                     "coordinates.name", "coordinates.id");
+
+        addVertexPropertyConstraints(jgm, PYPI_ARTIFACT_COORDINATES, "uuid", "version", "coordinates.extension",
+                                     "coordinates.name", "coordinates.build", "coordinates.abi", "coordinates.platform",
+                                     "coordinates.languageImplementationVersion", "coordinates.packaging",
+                                     "coordinates.distribution");
+
+        addVertexPropertyConstraints(jgm, ARTIFACT_TAG, "uuid");
+
+        addVertexPropertyConstraints(jgm, ARTIFACT_ID_GROUP, "uuid", "storageId", "repositoryId", "name");
+
+        addVertexPropertyConstraints(jgm, USER, "uuid", "username", "password",
+                                     "enabled", "roles", "securityTokenKey", "sourceId");
```

Comment:
Cosmetic: Could you please make all these one per line, where the list is longer than a line? Thanks! :)